pkg/handlers/event_stream.go

```diff
@@ -0,0 +1,161 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/optimizely/agent/pkg/middleware"
+	"github.com/optimizely/go-sdk/pkg/notification"
+	"log"
+	"net/http"
+)
+
+// A MessageChan is a channel of channels
+// Each connection sends a channel of bytes to a global MessageChan
+// The main broker listen() loop listens on new connections on MessageChan
+// New event messages are broadcast to all registered connection channels
+type MessageChan chan []byte
+
+// A EventStreamBroker holds open client connections,
+// listens for incoming events on its Notifier channel
+// and broadcast event data to all registered connections
+type EventStreamBroker struct {
+
+	// Events are pushed to this channel by the registered decision listener
+	Notifier chan notification.DecisionNotification
+
+	// New client connections
+	newClients chan MessageChan
+
+	// Closed client connections
+	closingClients chan MessageChan
+
+	// Client connections registry
+	clients map[MessageChan]bool
+}
+
+// Listen on different channels and act accordingly
+func (broker *EventStreamBroker) listen() {
```

Comment:
Per our conversation, the "broker" here is not necessary since we've elected to add new listeners with each connection. This approach puts the burden on the go-sdk to scalably handle listeners being added and removed and reliably subscribed to. Addressing the heavy lifting from that side will ensure a robust implementation on the go-sdk side. Effectively the NotificationCenter replaces the need for this Broker.